server/tests/lib/subject_page_config_test.py

```diff
@@ -42,58 +43,113 @@ def test(self):
             }
-        }
-        stat_var_spec {
-            key: "fire_area"
-            value {
-            stat_var: "Area_fireEvent"
-            name: "Area affected by fires"
+            stat_var_spec {
+                key: "filtered_1"
+                value {
+                    stat_var: "Area_fireEvent"
+                    denom: "Nonexistent"
+                }
             }
+            """, category)
+    stat_vars_existence = {
+        'variable': {
+            'Count_fireEvent': {
+                'entity': {
+                    'place_id': False
+                }
+            },
+            'Area_fireEvent': {
+                'entity': {
+                    'place_id': True
+                }
+            },
+            'TotalArea': {
+                'entity': {
+                    'place_id': True
+                }
+            },
+            'Nonexistent': {
+                'entity': {
+                    'place_id': False
+                }
+            },
         }
-        """, category)
-    expect = text_format.MessageToString(category)
+    }
+    result = lib_subject_page_config.exist_keys_category(
+        'place_id', category, stat_vars_existence)
+    expect = {
+        "kept_0": True,
+        "kept_1": True,
+        "filtered_0": False,
+        "filtered_1": False,
+    }
     assert result == expect
 
-    # Remove two more tiles, then the entire Fire category should be removed.
-    config = lib_subject_page_config.trim_config(
-        config, "Area_fireEvent", subject_page_pb2.Tile.TileType.HISTOGRAM)
-    config = lib_subject_page_config.trim_config(
-        config, "Area_fireEvent", subject_page_pb2.Tile.TileType.MAP)
-    assert config.categories[-1].title == "Floods"
+  @mock.patch('server.services.datacommons.observation_existence')
+  def test_remove_empty_charts(self, mock_observation_existence):
+
+    def obs_side_effect(all_svs, place_dcids):
+      return {
+          'variable': {
+              "sv_exists_1": {
+                  'entity': {
+                      'place_id': True
+                  }
+              },
+              "sv_exists_3": {
+                  'entity': {
+                      'place_id': True
+                  }
+              },
+              "sv_filtered_1": {
+                  'entity': {
+                      'place_id': False
+                  }
+              },
+              "sv_filtered_2": {
+                  'entity': {
+                      'place_id': False
+                  }
+              },
+              "sv_filtered_3": {
+                  'entity': {
+                      'place_id': False
+                  }
+              },
+          }
+      }
+
+    mock_observation_existence.side_effect = obs_side_effect
+    config = lib_util.get_subject_page_config(
+        "server/tests/test_data/existence.textproto")
+    result = lib_subject_page_config.remove_empty_charts(config, 'place_id')
+    print("result:")
```

Comment:
did you mean to keep these print statements?

Reply:
no i didn't - thanks, removed.